analytics/analytics/detectors/pattern_detector.py

```
@@ -49,12 +52,20 @@ def train(self, dataframe: pd.DataFrame, segments: list, cache: Optional[models.
             'cache': new_cache
         }
 
-    def detect(self, dataframe: pd.DataFrame, cache: Optional[models.ModelCache]) -> dict:
+    async def detect(self, dataframe: pd.DataFrame, cache: Optional[models.ModelCache]) -> dict:
         logger.debug('Unit {} got {} data points for detection'.format(self.analytic_unit_id, len(dataframe)))
         # TODO: split and sleep (https://github.com/hastic/hastic-server/pull/124#discussion_r214085643)
-        detected = self.model.detect(dataframe, self.analytic_unit_id, cache)
 
-        segments = [{ 'from': segment[0], 'to': segment[1] } for segment in detected['segments']]
+        window_size = cache.get('WINDOW_SIZE', 0)
```

I guess it would be better to throw an exception if `window_size` is undefined